plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -381,13 +401,6 @@ private function validate_options( $options ) {
 			return false;
 		}
 
-		// Hidden fields are not supported right now. They will be registered with hidden => false.
-		if ( ! empty( $options['hidden'] ) && true === $options['hidden'] ) {
-			$message = sprintf( 'Registering a field with hidden set to true is not supported. The field "%s" will be registered as visible.', $id );
-			_doing_it_wrong( '__experimental_woocommerce_blocks_register_checkout_field', esc_html( $message ), '8.6.0' );
-			// Don't return here unlike the other fields because this is not an issue that will prevent registration.
-		}
-
```

Comment:
Moved this down so it's more logical placement (it's an error that still allows adding the field).